src/libNode/FinalBlockProcessing.cpp

```diff
@@ -716,7 +716,11 @@ bool Node::ProcessFinalBlockCore(const bytes& message, unsigned int offset,
       LOG_GENERAL(WARNING, "StoreFinalBlock failed!");
       return false;
     }
-    if (!LOOKUP_NODE_MODE) {
+    // if lookup and loaded microblocks, then skip
+    lock_guard<mutex> g(m_mutexUnavailableMicroBlocks);
+    if (!(LOOKUP_NODE_MODE &&
+          m_unavailableMicroBlocks.find(txBlock.GetHeader().GetBlockNum()) !=
+              m_unavailableMicroBlocks.end())) {
```

Comment:
would this mean that we store new epochfin even if any of the microblocks are not received from shard?

Reply:
nope. this means we will store new epochfin if the txblock doesn't contain any microblock hash for the lookup to wait for.